### PR TITLE
#15 continuous integration using github workflows

### LINKED ISSUE #15: Continuous integration
Implement with github actions to be compliant with dynawo workflow

### PR DESCRIPTION
**Pull request by zamarrenolm**
closes #15

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -79,8 +80,13 @@ if (DYNAFLOW_LAUNCHER_SHARED_LIB)
 endif()
 message(STATUS "Compile internal libraries in " ${LINK_TYPE})
 
+
+
 set(CMAKE_CXX_FLAGS_DEBUG "-D_DEBUG_=1")
 set(CMAKE_CXX_FLAGS "-DLANG_CXX11")
+if(FORCE_CXX11_ABI)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_GLIBCXX_USE_CXX11_ABI=1")
```
Comment (lecourtoisflo) [inline CMakeLists.txt:88]: What the point of adding the ABI flag if all projects are compiled in the same container ?

Comment (zamarrenolm) [reply CMakeLists.txt:88]: It was proposed by @gautierbureau when we decided to prepare the builds for CI of dependant projects.

Comment (zamarrenolm) [reply CMakeLists.txt:88]: Maybe the container we are using requires it ?

Comment (lecourtoisflo) [reply CMakeLists.txt:88]: This flag is used for compatibility between old compilers and new compilers but if all are using the same version (i.e. the same container) I don't think it is required. TBC

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+jobs:
+  fedora:
+    name: Linux Fedora (Release, GCC, cxx11)
+    runs-on: ubuntu-latest
+    container: dynawo/dynawo-ci:latest
+    strategy:
+      fail-fast: false
+    env:
+      DYNAFLOW_LAUNCHER_BUILD_TYPE: Release
```
Comment (gautierbureau) [inline .github/workflows/ci.yml:23]: Should be Debug I think if we build tests

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+          cd dynawo-algorithms
+          unzip -qq "dynawo-algorithms.zip"
+          cd ..
+
+      - name: Load persistent build files
+        uses: actions/cache@v2
+        with:
+          # A list of files, directories, and wildcard patterns to cache and restore
+          path: |
+            build
```
Comment (gautierbureau) [inline .github/workflows/ci.yml:70]: Same remark for cache as the two other projects

File: .github/workflows/ci.yml (hunk 2 above)
Comment (zamarrenolm) [reply .github/workflows/ci.yml]: Changed build type to Debug

File: .github/workflows/ci.yml (hunk 3 above)
Comment (zamarrenolm) [reply .github/workflows/ci.yml]: Cache has been removed

File: CMakeLists.txt [hunk 4]
```diff
@@ -79,8 +80,13 @@ if (DYNAFLOW_LAUNCHER_SHARED_LIB)
 endif()
 message(STATUS "Compile internal libraries in " ${LINK_TYPE})
 
+
```
Comment (gautierbureau) [inline CMakeLists.txt:83]: useless blank line

File: CMakeLists.txt [hunk 5]
```diff
@@ -79,8 +80,13 @@ if (DYNAFLOW_LAUNCHER_SHARED_LIB)
...
+
```
Comment (gautierbureau) [inline CMakeLists.txt:84]: same

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Fetch Dynawo and Dynawo algorithms
+        run: |
+          source .github/workflows/tools.sh
+
+          download_asset() {
```
Comment (gautierbureau) [inline .github/workflows/ci.yml:39]: add it in ressource file

**Review by gautierbureau — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 4 above)
Comment (zamarrenolm) [reply CMakeLists.txt]: removed

**Comment by zamarrenolm** [comment]
> could you squash also when it is ok ? thanks

rebased on master and squashed all commits

**Review by gautierbureau — APPROVED** [review]

**Comment by gautierbureau** [comment]
The pipeline is successful !

**Pull request merged**